### DISCUSSION TITLE
Plans to Support Builder Pattern in Kube.rs?

### DISCUSSION THREAD

**Discussion by Oyelowo** [kube-rs/kube#1467] [Ideas]
Hello Kube.rs Community,

I’ve been exploring the kube-rs library and find it incredibly useful for interfacing with Kubernetes in Rust. However, I noticed that the library does not currently support the builder pattern through libraries like typed-builder or derive_builder, which can enhance the usability and type safety of building complex objects.

From my research across GitHub issues, discussions, and the Kube.rs Discord channel, I haven’t found any discussions or plans to implement this feature. Here are my questions:

1. Is there a specific reason why the builder pattern isn’t currently supported in Kube.rs? Understanding any underlying architectural or design considerations could provide valuable insights.

2.	Are there any plans to support the builder pattern in future releases? If not, could this be considered a valuable addition to the roadmap? The builder pattern can significantly ease the creation of objects with many optional parameters, which is common in Kubernetes configurations.

3.	How can the community contribute to such a feature? If there is interest in having builder pattern support, I would love to explore how I can contribute to the implementation.

Looking forward to the community’s insights and suggestions!

**Comment by clux** [comment]
1. No major reasons afaik. It more seems to be tradeoff in the size of k8s-openapi vs. the benefit it gives. I imagine it makes more sense if k8s-openapi implemented https://github.com/Arnavion/k8s-openapi/issues/77
2. It is [supported in kopium](https://github.com/kube-rs/kopium/blob/e9e3f884e1befb0483344de584d9cd118cf01178/src/main.rs#L45-L47), and we want to support it in [k8s-pb](https://github.com/kube-rs/k8s-pb) - old issue in https://github.com/kube-rs/k8s-pb/issues/9
3. Any help on the codegen side is appreciated, although if we are talking about k8s-pb that might take some time before it's in a ready state (am planning on writing up an issue for the integration of this soon™)
